plugins/com.google.cloud.tools.eclipse.sdk/src/com/google/cloud/tools/eclipse/sdk/internal/CloudSdkPreferences.java

```diff
@@ -57,7 +61,42 @@ static IEclipsePreferences getPreferenceNode() {
 
   @Override
   public void initializeDefaultPreferences() {
-    getPreferenceStore().setDefault(CLOUD_SDK_MANAGEMENT,
-        CloudSdkManagementOption.AUTOMATIC.name());
+    initializeDefaultPreferences(getPreferenceStore());
+  }
+
+  @VisibleForTesting
+  public void initializeDefaultPreferences(IPreferenceStore preferences) {
+    if (CloudSdkManager.isManagedSdkFeatureEnabled()) {
+      if (!preferences.contains(CLOUD_SDK_MANAGEMENT)) {
+        // If the CLOUD_SDK_MANAGEMENT preference has not been set, then determine the
+        // appropriate setting. Note that CloudSdkPreferenceResolver only checks for
+        // the Managed Cloud SDK when CLOUD_SDK_MANAGEMENT has been explicitly set
+        // (i.e., this code has been run).
+        configureManagementPreferences(preferences, isCloudSdkAvailable());
+      }
+      preferences.setDefault(CLOUD_SDK_MANAGEMENT, CloudSdkManagementOption.AUTOMATIC.name());
+    }
+  }
+
+  /** Configure the managed SDK settings given current settings. */
+  @VisibleForTesting
+  static void configureManagementPreferences(
+      IPreferenceStore preferences, boolean cloudSdkAvailable) {
+    // has the user previously set the Cloud SDK path? has it been found in a well-known location?
+    if (preferences.contains(CLOUD_SDK_PATH) || cloudSdkAvailable) {
```

Comment:
For `CLOUD_SDK_PATH`, I think it's much safer and better to check `! ... .isEmpty()` instead. Leaving the field empty doesn't feel like the user has actually set the path. In this case, I think it's nicer to auto-download the SDK if it is locally not available.

Reply:
ok, fair enough.